core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -1519,6 +1520,7 @@ public ServerBuilder decoratorUnder(String prefix,
     @UnstableApi
     public ServerBuilder errorHandler(ServerErrorHandler errorHandler) {
         this.errorHandler = requireNonNull(errorHandler, "errorHandler");
```

Comment:
How about removing `ServerBuilder.errorHandler` field and always using `virtualHostTemplate` to keep the server-level error handler?

You'll need to make the following two changes as well:

1. Update the constructor of `ServerBuilder` to set the default:
   ```
   ServerBuilder() {
       ...
       virtualHostTemplate.errorHandler(ServerErrorHandler.ofDefault());
   }
   ```
2. Add the `VirtualHost.errorHandler()` getter.
3. Deprecate `ServerConfig.errorHandler()` in favor of `VirtualHost.errorHandler()`.

(If not sure, please take a look at `ServerBuilder.accessLogWriter()` and `VirtualHost.accessLogWriter()`.)